src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionsList.tsx

```diff
@@ -52,7 +52,7 @@ const ProposedSolutionsList: React.FC<ProposedSolutionsListProps> = ({ proposedS
           <ProposedSolutionView key={i} proposedSolution={proposedSolution} crReviewed={crReviewed} />
         ))}
       </div>
-      {crReviewed === undefined && auth.user?.role !== 'GUEST' ? (
+      {crReviewed === undefined && (auth.user ? !isGuest(auth.user.role) : false) ? (
```

Comment:
what if we just made all these functions possibly take in undefined and if it's undefined it returns false or whatever